agenthub/codeact_agent/codeact_agent.py

```diff
@@ -219,6 +219,12 @@ def step(self, state: State) -> Action:
                     raise NotImplementedError(
                         f'Unknown observation type: {obs.__class__}'
                     )
+            self.messages.append(
+                {
+                    'role': 'user',
+                    'content': f'ENVIRONMENT REMINDER: You have {state.max_iterations - state.iteration - 1} turns left to complete the task.',
```

Comment:
Can we concatenate this after the existing environment observation? I remember some LLM API does not work if you put two consecutive user messages.

Reply:
👍 sg! we might want to look into this more closely--I think codeact can end up with consecutive user messages